### DISCUSSION TITLE
fluentd && grepcounter && also with logging

### DISCUSSION THREAD

**Discussion by netseevol** [fluent/fluentd#4365] [Q&A]
Hi All.

Having issue with fluentd and grepcounter and mail. My setup docker that is using fluend as logging driver and then is plan to send log into file(s) for each container. That works fine, but if I want to use grepcounter and mail, then I cannot get log into file.

basic logging works like this and it works fine.

<match taggedservice1.**>
 @type copy
 <store>
        @type file
        path /var/log/td-agent/app/taggedservice1
        time_slice_format %Y%m%d%H
        time_slice_wait 10m
        time_format %Y%m%dT%H%M%S%z
        compress gzip
        utc
 </store>
 <store>
        @type stdout
 </store>
</match>

This works fine, but If I add grepcounter, then I will lose ability to log into file like this:

<match taggedservice1.**>
   @type grepcounter
   count_interval 10 # The time window for counting errors (in secs)
   input_key message # The field to apply the regular expression
   regexp "Action send failed"
   threshold 2   # The minimum number of errors to trigger an alert
   add_tag_prefix catastrofic_error # Generate tags like "error_5xx.apache.access"#
 <store>
        @type stdout
 </store>

</match>

<match catastrofic_error.taggedservice1.**>
@type copy
<store>
  @type mail
  host 127.0.0.1 # Change this to your SMTP server host
  port 25 # Normally 25/587/465 are used for submission
  enable_starttls_auto true # Use this option to enable STARTTLS
  from xxx@xxx.com # Set the sender address
  to xxx@xxxx.com # Set the recipient address
  subject  ‘service failed multiple times ()’
  message Total Service error count: %s\n\nPlease check environment ASAP
  message_out_keys count # Use the "count" field to replace "%s" above
</store>
 <store>
        @type stdout
 </store>

</match>

So, when using grepcounter I will lose ability to get log into files and same time email alerts. any workarounds for that issue? 
Sorry for stupid question..

Eero

**Comment by swapwarule19** [comment]
Yes, use this store and copy plugin to get over it. I used this, it allows both file saving as well as email.

 
![image](https://github.com/fluent/fluentd/assets/37341115/588510a9-898b-4df0-9f40-46bda31c83f0)
